torch/testing/_internal/common_utils.py

```diff
@@ -1038,6 +1040,34 @@ def __torch_function__(self, func, types, args=(), kwargs=None):
 TEST_WITH_AOT_EAGER = os.getenv('PYTORCH_TEST_WITH_AOT_EAGER') == '1'
 TEST_WITH_TORCHDYNAMO = os.getenv('PYTORCH_TEST_WITH_DYNAMO') == '1' or TEST_WITH_TORCHINDUCTOR or TEST_WITH_AOT_EAGER
 
+
+# Returns a string prefix usable to set environment variables for any test
+# features (set by TEST_WITH_*) that are enabled during this instantiation
+# of the test suite.
+# Example: "PYTORCH_TEST_WITH_ASAN=1 PYTORCH_TEST_WITH_ROCM=1"
+def get_test_feature_env_var_prefix():
+    # To keep the output minimal, track which flags imply other flags.
+    implications = {
+        "TEST_WITH_TORCHINDUCTOR": {"TEST_WITH_TORCHDYNAMO"},
+        "TEST_WITH_AOT_EAGER": {"TEST_WITH_TORCHDYNAMO"},
+    }
+
+    enabled = set({flag for flag, on in globals().items() if flag.startswith("TEST_WITH_") and on})
+    implied = set({})
+    [implied.update(implications[flag]) for flag in enabled if flag in implications]
+    enabled = [flag for flag in enabled if flag not in implied]
+
+    # These don't follow the env var name template of PYTORCH_<var>.
+    non_standard_mappings = {
+        "TEST_WITH_TORCHINDUCTOR": "PYTORCH_TEST_WITH_INDUCTOR",
+        "TEST_WITH_TORCHDYNAMO": "PYTORCH_TEST_WITH_DYNAMO",
+    }
+
+    return " ".join([f"{non_standard_mappings[flag]}=1"
+                     if flag in non_standard_mappings
+                     else f"PYTORCH_{flag}=1" for flag in enabled])
```

Comment:
Since this is a BE patch, I'm going to ask you to do a little more surgery. Instead of hacking up globals() to detect environment variable settings, can we instead move the entire test environment variable apparatus to a class that can keep track of all the test knobs? If your primary objective is to avoid having to edit use sites of things like TEST_WITH_TORCHINDUCTOR, you can still assign these globals; the idea is to just have a clear class that handles *defining* the variables in one place.

Reply:
Cool, I'm on-board with this. I actually started down this route initially but decided to keep it simple / hacky in the initial commit.

Reply:
This could easily fall prey to scope creep trying to gather all test configuration into a single class. For now, I introduced a simple `TestEnvironment` class that handles environment variable related boolean flags, exporting flags globally to avoid call-site changes. I could also extend it to:
1. Handle non-boolean environment variable settings
2. Handle all test related flags that aren't related to environment variables

We'd be introducing more complexity and changes for 1 and especially for 2. Neither of those are strictly necessary for the repro command use case, and they'd have questionable payoff. wdyt?